sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfiguration.java

```diff
@@ -133,6 +132,17 @@ public static StructuredConfigProperties toConfigProperties(OpenTelemetryConfigu
     return toConfigProperties((Object) model);
   }
 
+  /**
+   * Convert the {@code configuration} YAML to a generic {@link StructuredConfigProperties}.
+   *
+   * @param configuration configuration YAML
+   * @return a generic {@link StructuredConfigProperties} representation of the model
+   */
+  public static StructuredConfigProperties toConfigProperties(InputStream configuration) {
+    Object yamlObj = loadYaml(configuration, System.getenv());
+    return toConfigProperties(yamlObj);
+  }
```

Comment:
This is an essential test utility, allowing you to write chunks of YAML, convert them to StructuredConfigProperties, and call ComponentProvider, i.e. https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1440/files#diff-3d0118f231281243bf34a7e7beb3809e7a6523e8ece8a17b83f1784953389ceb